src/credentials/FabricTable.h

```diff
@@ -1027,6 +1029,18 @@ class DLL_EXPORT FabricTable
      */
     CHIP_ERROR SetFabricIndexForNextAddition(FabricIndex fabricIndex);
 
+    /**
+     * @brief Set the advertising behavior for the fabric identified by `fabricIndex`.
+     *
+     * It is the caller's responsibility to actually restart DNS-SD advertising
+     * as needed after updating this state.
+     *
+     * @param fabricIndex - Fabric Index for which to set the label
+     * @param advertiseIdentity - whether the identity for this fabric should be advertised.
+     * @retval CHIP_ERROR_INVALID_FABRIC_INDEX if fabricIndex does not refer to a fabric in the table
+     */
+    CHIP_ERROR SetShouldAdvertiseIdentity(FabricIndex fabricIndex, AdvertiseIdentity advertiseIdentity);
```

Comment:
Please test the getter in FabricInfo in the unit test after several cases of SetShouldAdveritseIdentity.

Reply:
Can do.